src/onnx/parse_slice.cpp

```diff
@@ -77,7 +78,37 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        std::vector<int64_t> raxes;
+        if(std::any_of(steps.begin(), steps.end(), [](auto s) { return s < 0; }))
```

Comment:
You dont need this branch, as the for loops will skip when `raxes` is empty.

Reply:
I had to add this, otherwise `logsoftmax_nonstd_input_test` kept segfaulting:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check: __n (which is 3) >= this->size() (which is 1)
```

Reply:
I think thats masking a bug.